### PR TITLE
fix: does to stop upgrade because was impossible to remove rook

### PR DESCRIPTION
**Pull request by camilamacedo86**
#### What this PR does / why we need it:

If in the upgrade we be unable to remove Rook we can still be able to do that manually afterwords or by re-calling the script. However, does not seems make sense we stop the cluster upgrade because did when we could check that we migrated the PVC(s), the ObjectStore successfully. 

#### Which issue(s) this PR fixes:

Fixes # [sc-70239]

#### Does this PR introduce a user-facing change?

```release-note
Fixes script stop when the migration from Rook were done successfully but only was not possible to remove Rook. Note that it still possible to call the task `curl <installer>/task.sh | sudo bash -s remove_rook_ceph` to remove Rook manually or just re-run the script to re-try.
```

### CODE REVIEW

File: scripts/common/rook.sh [hunk 1]
```diff
@@ -216,7 +216,10 @@ function maybe_cleanup_rook() {
 
         if [ "$DID_MIGRATE_ROOK_PVCS" == "1" ] && [ "$DID_MIGRATE_ROOK_OBJECT_STORE" == "1" ]; then
             report_addon_start "rook-ceph-removal" "v1"
-            remove_rook_ceph
+            if ! remove_rook_ceph; then
+                logFail "Unable to remove Rook."
+                return
+            fi
             report_addon_success "rook-ceph-removal" "v1"
```
Comment (emosbaugh) [inline scripts/common/rook.sh:223]: it does not seem like we should report successful here but there does not see any way to report a failure. @laverya what do you recommend?

Comment (camilamacedo86) [reply scripts/common/rook.sh:223]: That is a good point.

Comment (laverya) [reply scripts/common/rook.sh:223]: I swear we had a report_addon_failure command

Comment (laverya) [reply scripts/common/rook.sh:223]: congratulations, you can be the first user
https://github.com/replicatedhq/kURL/blob/4d146f4e2ac810f92376100041e2c6388e016203/scripts/common/reporting.sh#L144-L166
just make sure to get rid of the 'unused' comment 😆

Comment (laverya) [reply scripts/common/rook.sh:223]: actually that's probably not quite what you want
maybe make a copy of the https://github.com/replicatedhq/kURL/blob/4d146f4e2ac810f92376100041e2c6388e016203/scripts/common/reporting.sh#L104 function but with the fail_addon endpoint instead

File: scripts/common/rook.sh [hunk 2]
```diff
@@ -216,7 +216,11 @@ function maybe_cleanup_rook() {
 
         if [ "$DID_MIGRATE_ROOK_PVCS" == "1" ] && [ "$DID_MIGRATE_ROOK_OBJECT_STORE" == "1" ]; then
             report_addon_start "rook-ceph-removal" "v1"
-            remove_rook_ceph
+            if ! remove_rook_ceph; then
+                logFail "Unable to remove Rook."
+                report_addon_fail "rook-ceph-removal" "v1"
```
Comment (emosbaugh) [inline scripts/common/rook.sh:221]: does this function exist?

**Review by emosbaugh — CHANGES_REQUESTED** [review]

File: scripts/common/rook.sh (hunk 2 above)
Comment (laverya) [reply scripts/common/rook.sh:221]: I like the new function

**Review by laverya — APPROVED** [review]

**Pull request merged**